### DISCUSSION TITLE
Is it possible to add a selected state to an element in the Stencil?

### DISCUSSION THREAD

**Discussion by tkaabbc** [clientIO/joint#2709] [Q&A]
### Introduction

Hi team,
I need to implement a feature that will drop a **Stencil** element multiple times after it has been selected, so I need to give the selected element a selected state, e.g. change the background colour.

For example, **add a red background when selected** as shown below. **When clicking the selected element again, it needs to deselect**, which means removing the background color.

The method I used is to get the coordinates of the clicked element and add a red background `<rect fill="red" stroke="red">` at the coordinates. However, this feels a bit messy and will affect other Stencil functionalities, while requiring a lot of conditional checks. I’m not sure if there is a better way?🙏
![Xnip2024-07-17_23-34-26](https://github.com/user-attachments/assets/d9ad731e-717e-4c0d-9575-ce5587b72aa2)

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS+

**Comment by kumilingus** [comment]
Duplicate of https://github.com/clientIO/joint/discussions/2605 and https://github.com/clientIO/joint/discussions/2535.
